submodule_test.go

```diff
@@ -15,16 +14,16 @@ import (
 type SubmoduleSuite struct {
 	BaseSuite
 	Worktree *Worktree
-	path     string
+	clean    func()
 }
 
 var _ = Suite(&SubmoduleSuite{})
 
 func (s *SubmoduleSuite) SetUpTest(c *C) {
 	path := fixtures.ByTag("submodule").One().Worktree().Root()
 
-	dir, clean := s.TemporalDir()
-	defer clean()
```

Comment:
This was also a bug (but wasn't causing any issues because `clean()` wasn't actually cleaning. We don't want to clean at the end of the set up, we want to clean after the test has finished (during tear down)